sdk/src/wallet/migration/chrysalis.rs

```diff
@@ -0,0 +1,314 @@
+                        native_token_foundries: HashMap::new(),
+                    })
+                }
+            }
+        }
+    }
+    // Accounts must be ordered by index
+    new_accounts.sort_unstable_by_key(|a| a.index);
+    Ok((new_accounts, secret_manager_dto))
+}
+
+fn get_chrysalis_data(chrysalis_storage_path: &Path, password: Option<Password>) -> Result<HashMap<Vec<u8>, String>> {
+    let encryption_key = password.map(storage_password_to_encryption_key);
+    let chrysalis_db = DB::open_default(chrysalis_storage_path)?;
+    let mut chrysalis_data = HashMap::new();
+    // iterate over all rocksdb keys
+    for item in chrysalis_db.iterator(IteratorMode::Start) {
+        let (key, value) = item?;
+
+        let key_utf8 = String::from_utf8(key.to_vec()).map_err(|_| Error::Migration("invalid utf8".into()))?;
+        let value = if let Some(encryption_key) = &encryption_key {
+            let value_utf8 = String::from_utf8(value.to_vec()).map_err(|_| Error::Migration("invalid utf8".into()))?;
+            // "iota-wallet-key-checksum_value" is never an encrypted value
+            if key_utf8 == "iota-wallet-key-checksum_value" {
+                value_utf8
+            } else if let Ok(value) = serde_json::from_str::<Vec<u8>>(&value_utf8) {
+                decrypt_record(value, encryption_key)?
+            } else {
+                value_utf8
+            }
+        } else {
+            String::from_utf8(value.to_vec()).map_err(|_| Error::Migration("invalid utf8".into()))?
+        };
+
+        chrysalis_data.insert(key.to_vec(), value);
+    }
+    Ok(chrysalis_data)
+}
+
+fn storage_password_to_encryption_key(password: Password) -> Zeroizing<[u8; 32]> {
+    let mut dk = [0; 64];
+    // safe to unwrap (rounds > 0)
+    crypto::keys::pbkdf::PBKDF2_HMAC_SHA512(
+        password.as_bytes(),
+        b"wallet.rs::storage",
+        core::num::NonZeroU32::new(100).unwrap(),
+        &mut dk,
+    );
+    let key: [u8; 32] = dk[0..32][..].try_into().unwrap();
+    Zeroizing::new(key)
+}
+
+fn decrypt_record(record_bytes: Vec<u8>, encryption_key: &[u8; 32]) -> crate::wallet::Result<String> {
+    let mut record: &[u8] = &record_bytes;
+
+    let mut nonce = [0; XChaCha20Poly1305::NONCE_LENGTH];
+    record.read_exact(&mut nonce)?;
+
+    let mut tag = vec![0; XChaCha20Poly1305::TAG_LENGTH];
+    record.read_exact(&mut tag)?;
+
+    let mut ct = Vec::new();
+    record.read_to_end(&mut ct)?;
+
+    let mut pt = vec![0; ct.len()];
+    // we can unwrap here since we know the lengths are valid
+    XChaCha20Poly1305::decrypt(
+        encryption_key.try_into().unwrap(),
+        &nonce.try_into().unwrap(),
+        &[],
+        &mut pt,
+        &ct,
+        tag.as_slice().try_into().unwrap(),
+    )
+    .map_err(|e| Error::Migration(format!("{:?}", e)))?;
+
+    Ok(String::from_utf8_lossy(&pt).to_string())
+}
+
+pub(crate) fn key_to_chrysalis_key(key: &[u8], stronghold: bool) -> Vec<u8> {
```

Comment:
Should be private probably. Also can we name it a little better? Maybe `stardust_key_to_chrysalis` or just `to_chrysalis_key`